### ISSUE TITLE
selector-max-empty-lines

### ISSUE THREAD

**Issue by dryoma** [stylelint/stylelint#1142]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Not sure if these are bugs, but could cause problems. There are situations where extra blank lines are accepted by stylelint, which is in most cases undesirable. The only way of preventing this I found is to set `"max-empty-lines": 0`

> Which rule, if any, is this issue related to?

"max-empty-lines"
"selector-list-comma-newline-after"
"block-opening-brace-newline-after"

> What CSS is needed to reproduce this issue?

```css
// 1) Both are technically correct, yet both clearly breach the code style
.u-myComp,

.U-myComp {
  ...
}

.u-myComp,
foo
.U-myComp {
  ...
}
// 2) There is no way to force no extra line here. Except setting `max-empty-lines` to 0, which is often undesirable.
.U-myComp {

  margin: 0
}
```
There might be other cases.

> What stylelint configuration is needed to reproduce this issue?

The relevant rules are:
```json
{
  "rules": {
    "max-empty-lines": 1,
    "selector-list-comma-newline-after": "always",
    "block-opening-brace-newline-after": "always-multi-line",
  }
}
```

> Which version of stylelint are you using?

`6.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

```js
.pipe(gulpPostCSS([
  // nested(),
  stylelint(),
  reporter({
    clearMessages: true,
    throwError: true
  })
], {
  syntax: scssSyntax
}))
```

> What did you expect to happen?

No extra blank lines or some settings to disallow them except `"max-empty-lines": 1`

> What actually happened (e.g. what warnings or errors you are getting)?

**Comment by davidtheclark** [comment]
We recently introduced [`function-max-empty-lines`](http://stylelint.io/user-guide/rules/function-max-empty-lines/), which I think does what you're looking for: provides a more granular way of reducing empty lines in specific situations. So we could come up with more rules to do the same in different situations.

For example, to address:

```
.u-myComp,

.U-myComp {
  ...
}
```

We could make `selector-max-empty-lines` maybe.

How does that sound?

**Comment by jeddy3** [comment]
> How does that sound?
Ha, I was just about to post the exact same thing (delete that bit now)... 

Limiting the number of empty lines *within declaration blocks* is tricker though as the [`declaration-block-properties-order`](https://github.com/stylelint/stylelint/tree/master/src/rules/declaration-block-properties-order) rules allows declarations to be grouped together, and there's [a proposal](https://github.com/stylelint/stylelint/issues/752) for limiting the number of empty lines *within this declaration **groups*** (but not *between groups*). I'll open a separate issue to discuss that so this one can be focused on selectors.

**Comment by jeddy3** [comment]
- Name: `selector-max-empty-lines`
- Primary option: `int`
- Secondary options: none
- Message: "Unexpected empty line within selector"
- Section: "Selector"

Sound good? @dryoma Does this sound like a rule you'd like to champion (i.e. help us build)?

**Comment by evilebottnawi** [comment]
Merged. Close.

**Comment by dryoma** [comment]
Thanks!

**Comment by evilebottnawi** [comment]
@dryoma :beers:

**Comment by ntwb** [comment]
When closing issues that have been resolved via a pull request please mention the commit it relates to when closing the issue, again, it makes it easier to track for everyone 😄 

Merged via https://github.com/stylelint/stylelint/commit/2ad7027cda6cbd3e168831ef422f72ea12fa9c50
